### PR TITLE
Add MET09 IODC service mode

### LINKED ISSUE #2270: Missing SEVIRI service mode for new IODC sub-satellite longitude of MET09
**Describe the bug**
The new service mode of MET09 (IODC at 45.5 degrees East) is not included in https://github.com/pytroll/satpy/blob/df8ddd1897703cc466bfaab109c58c003e120f68/satpy/readers/eum_base.py#L89 leading to errors when reading SEVIRI L2 bufr data with area definition (`reader_kwargs={'with_area_definition': True}`)

**To Reproduce**

```python
from satpy import Scene

sev_l2_bufr_file = '<filename>'
scn = Scene(filenames=[sev_l2_bufr_file], reader='seviri_l2_bufr', reader_kwargs={'with_area_definition': True)
scn.load(['ki'])

```

**Expected behavior**
Successfully load the data and resample onto the original 2D grid.

**Actual results**
```
INFO yaml_reader.py Line 1005: Dataset longitude is missing the area attribute and will not be flipped.
INFO yaml_reader.py Line 1005: Dataset latitude is missing the area attribute and will not be flipped.
WARNING yaml_reader.py Line 700: Failed to load DataID(name='ki', resolution=9001.209497451, modifiers=()) from <SeviriL2BufrFileHandler: '/tcenas/fbf/mpef/store/OPEB/GIIBUFRProduct/GIIBUFRProduct_20221101120000Z_00_OMPEFS02_MET09_FES_E0455'>
Traceback (most recent call last):
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/yaml_reader.py", line 695, in _load_dataset
    projectable = fh.get_dataset(dsid, ds_info)
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/seviri_l2_bufr.py", line 204, in get_dataset
    xarr = self.get_dataset_with_area_def(arr, dataset_id)
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/seviri_l2_bufr.py", line 226, in get_dataset_with_area_def
    self._area_def = self._construct_area_def(dataset_id)
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/seviri_l2_bufr.py", line 267, in _construct_area_def
    stand_area_def = get_area_def(area_naming['area_id'])
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/resample.py", line 223, in get_area_def
    return parse_area_file(get_area_file(), area_name)[0]
  File "/tcenas/proj/optcalimg/strandgren/miniconda3/envs/pysat_latest/lib/python3.8/site-packages/pyresample/area_config.py", line 138, in parse_area_file
    return _parse_yaml_area_file(area_file_name, *regions)
  File "/tcenas/proj/optcalimg/strandgren/miniconda3/envs/pysat_latest/lib/python3.8/site-packages/pyresample/area_config.py", line 191, in _parse_yaml_area_file
    raise AreaNotFound('Area "{0}" not found in file "{1}"'.format(area_name, area_file_name))
pyresample.area_config.AreaNotFound: 'Area "msg_seviri_unknown_9km_ext" not found in file "[\'/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/etc/areas.yaml\', \'/tcenas/home/strandgren/git/eum/py/local_readers/areas.yaml\']"'
ERROR yaml_reader.py Line 849: Could not load dataset 'DataID(name='ki', resolution=9001.209497451, modifiers=())': "Could not load DataID(name='ki', resolution=9001.209497451, modifiers=()) from any provided files"
Traceback (most recent call last):
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/yaml_reader.py", line 847, in _load_dataset_with_area
    ds = self._load_dataset_data(file_handlers, dsid, **kwargs)
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/yaml_reader.py", line 719, in _load_dataset_data
    proj = self._load_dataset(dsid, ds_info, file_handlers, **kwargs)
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/readers/yaml_reader.py", line 704, in _load_dataset
    raise KeyError(
KeyError: "Could not load DataID(name='ki', resolution=9001.209497451, modifiers=()) from any provided files"
WARNING scene.py Line 1276: The following datasets were not created and may require resampling to be generated: DataID(name='ki', resolution=9001.209497451, modifiers=())
INFO view.py Line 76:  Generating plotting window and visualizing requested satellite data.
Traceback (most recent call last):
  File "/tcenas/home/strandgren/git/ext/py/satpy/latest/satpy/satpy/dataset/data_dict.py", line 169, in __getitem__
    return super(DatasetDict, self).__getitem__(item)
KeyError: 'ki'

```

**Additional context**
To be fixed by @strandgren during PCW in Athens. This fix will also include modifying the MSG IODC area definitions in `areas.yaml` to match the current sub-satellite longitude of MET09 (45.5 degrees North). This will be a temporary work-around until #2238 has been realized.

### PR DESCRIPTION
**Pull request by strandgren**
This PR adds a new service mode of SEVIRI, which is the IODC mission at 45.5 degrees East. This is the position of the MET09 satellite which took over the operational IODC service in summer 2022. 

Furthermore, this PR updates the sub-satellite longitude of the standardized area definitions of the MSG SEVIRI IODC mission. This is needed in order to be compliant with current/operational data until issue #2238 has been resolved.

 - [x] Closes #2270 
 - [x] Tests added/modified

### CODE REVIEW

**Comment by pnuu** [comment]
The tests need some adjustments in `satpy.tests.reader_tests.test_seviri_l2_bufr.TestSeviriL2BufrReader.test_data_with_area_definition`: 
- different `lon_0`
- minor difference in extents (rounding?)

**Comment by strandgren** [comment]
ups, given how minor the change was I forgot that the tests were affected.. Have fixed this now. The differences were indeed rounding errors and have also been fixed. Thanks for reviewing!

**Review by pnuu — APPROVED** [review]
LGTM

**Review by sfinkens — APPROVED** [review]
Thanks a lot!

File: satpy/readers/eum_base.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def get_service_mode(instrument_name, ssp_lon):
     """Get information about service mode for a given instrument and subsatellite longitude."""
     service_modes = {'seviri': {'0.0':  {'service_name': 'fes', 'service_desc': 'Full Earth Scanning service'},
                                 '9.5':  {'service_name': 'rss', 'service_desc': 'Rapid Scanning Service'},
-                                '41.5': {'service_name': 'iodc', 'service_desc': 'Indian Ocean Data Coverage service'}
+                                '41.5': {'service_name': 'iodc', 'service_desc': 'Indian Ocean Data Coverage service'},
+                                '45.5': {'service_name': 'iodc', 'service_desc': 'Indian Ocean Data Coverage service'}
```
Comment (sjoro) [inline satpy/readers/eum_base.py:94]: Please add a test in `eum_base` for this new service.

**Review by sjoro — CHANGES_REQUESTED** [review]
A small test should be added to `eum_base` to cover the new service location.

File: satpy/readers/eum_base.py (hunk 1 above)
Comment (strandgren) [reply satpy/readers/eum_base.py:94]: Added.

**Review by sjoro — APPROVED** [review]

**Pull request merged**